lib/mini_autobot/google_sheets.rb

```diff
@@ -0,0 +1,77 @@
+module MiniAutobot
+
+  # Class that supports integration with Google Sheets
+  # See documentation here: https://developers.google.com/drive/v3/web/about-auth to set up OAuth 2.0
+  # access to your sheets. Place the info in google_drive_config.json in your config/mini_autobot folder
+  #
+  # To use, use the -g or --google_sheets parameter with the id of your sheet
+  # Example: -g 1gWbpPq7rrTdNtSJSN2ljDlPE-okf0w3w7ai5PlhY8bk
+  class GoogleSheets
+    require 'rubygems'
+    require 'google/api_client'
+    require 'google_drive'
+    require 'json'
+
+    def initialize
+      @session = session
+      @spreadsheet = spreadsheet
+      @worksheet = worksheet
+      @automation_serial_column = automation_serial_column
+      @selected_browser_column = selected_browser_column
+    end
+
+    def session
+      GoogleDrive.saved_session(MiniAutobot.root.join('config/mini_autobot', 'google_drive_config.json'))
+    end
+
+    def spreadsheet
+      @session.spreadsheet_by_key(MiniAutobot.settings.google_sheet)
```

Comment:
Consider separating your data from your logic in cases like `#spreadsheet`. The `GoogleSheets` class shouldn't rely on a global state, which `MiniAutobot.settings` is.

Instead, one design choice could be to have the spreadsheet name passed in to the class, e.g., when instantiating, consider the incantation:

```ruby
sheet = GoogleSheets.new(MiniAutobot.settings.google_sheet)
```

In fact, the same standard can be applied to the `#session` method above.